File: src/main/java/com/adjorno/billib/rest/model/TrackInfo.java
package com.adjorno.billib.rest.model;

import com.adjorno.billib.rest.db.Track;

import java.util.Map;

public class TrackInfo {
    private Track mTrack;

    private Map<String, Map<String, Integer>> mHistory;

    private long mGlobalRank;

    public TrackInfo() {
    }

    public Track getTrack() {
        return mTrack;
    }

    public void setTrack(Track track) {
        mTrack = track;
    }

    public Map<String, Map<String, Integer>> getHistory() {
        return mHistory;
    }

    public void setHistory(Map<String, Map<String, Integer>> history) {
        mHistory = history;
    }

    public long getGlobalRank() {
        return mGlobalRank;
    }

    public void setGlobalRank(long globalRank) {
        mGlobalRank = globalRank;
    }
}

File: backend/src/main/java/com/ifochka/m14n/rest/model/TrackInfo.kt
package com.ifochka.m14n.rest.model

import com.ifochka.m14n.rest.db.Track

data class TrackInfo(
    var track: Track? = null,
    var history: Map<String, Map<String, Int>>? = null,
    var globalRank: Long = 0,
)
